New Resource: `azurerm_spring_cloud_service` (#4928)
terraform-providers/terraform-provider-azurerm: website/docs/r/spring_cloud_service.html.markdown

  name     = "example-resources"
  location = "Southeast Asia"
}

resource "azurerm_spring_cloud_service" "example" {
  name                = "example-springcloud"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  config_server_git_settings {
    uri          = "https://github.com/Azure-Samples/piggymetrics"
    label        = "config"
    search_paths = ["dir1", "dir2"]
  }

  tags = {
    Env = "staging"
  }
}
```

## Argument Reference

The following arguments are supported:

* `name` - (Required) Specifies the name of the Spring Cloud Service resource. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) Specifies The name of the resource group in which to create the Spring Cloud Service. Changing this forces a new resource to be created.

* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.

-> **Note:** At this time Azure Spring Cloud Service is only supported in a subset of regions (including `East US`, `South East Asia`, `West Europe` and `West US 2`.

> * `config_server_git_settings` - (Optional) One or more `config_server_git_setting` block defined below.

Reviewer: ```suggestion
* `config_server_git_settings` - (Optional) A `config_server_git_setting` block as defined below.
```

* `tags` - (Optional) A mapping of tags to assign to the resource.

---

The `config_server_git_setting` block supports the following:

* `uri` - (Required) The URI of the default Git repository used as the Config Server back end, should be started with `http://`, `https://`, `git@`, or `ssh://`.

* `label` - (Optional) The default label of the Git repository, should be the branch name, tag name, or commit-id of the repository.

* `search_paths` - (Optional) An array of strings used to search subdirectories of the Git repository.

> * `http_basic_auth` - (Optional) One or more `http_basic_auth` block defined below.

Reviewer: ```suggestion
* `http_basic_auth` - (Optional) An `http_basic_auth` block as defined below.
```

> * `ssh_auth` - (Optional) One or more `ssh_auth` block defined below.

Reviewer: ```suggestion
* `ssh_auth` - (Optional) An `ssh_auth` block as defined below.
```

> * `repositories` - (Optional) One or more `repository` block defined below.

Reviewer: Again, should be singular name
```suggestion
* `repository` - (Optional) One or more `repository` blocks as defined below.
```

---

The `repository` block supports the following:

* `name` - (Required) A name to identify on the Git repository, required only if repos exists.

* `uri` - (Required) The URI of the Git repository that's used as the Config Server back end should be started with `http://`, `https://`, `git@`, or `ssh://`.

* `pattern` - (Optional) An array of strings used to match an application name. For each pattern, use the `{application}/{profile}` format with wildcards.

* `label` - (Optional) The default label of the Git repository, should be the branch name, tag name, or commit-id of the repository.

* `search_paths` - (Optional) An array of strings used to search subdirectories of the Git repository.

> * `http_basic_auth` - (Optional) One or more `http_basic_auth` block defined below.

Reviewer: Max 1
```suggestion
* `http_basic_auth` - (Optional) An `http_basic_auth` block as defined below.
```

> * `ssh_auth` - (Optional) One or more `ssh_auth` block defined below.

Reviewer: Max 1
```suggestion
* `ssh_auth` - (Optional) An `ssh_auth` block as defined below.
```

> subcategory: "App Platform"

Author: @jackofallops just curious about this rule, because in my previous comments, Tom told me to remove the first line.  https://github.com/terraform-providers/terraform-provider-azurerm/pull/4928#discussion_r394643438
so in my future PRs, should I add the first line `---` ?
